### ISSUE TITLE
[FR] Support of more HuggingFace embedders for multimodality

### ISSUE THREAD

**Issue by eostis** [vespa-engine/vespa#28090]
The goal is multi-modal search, which requires multi-modal embedders.

For instance, E-commerce can use:
- text-to-image (CLIP): search images
- text-to-text: search texts
- image-to-image: similar images.
- 
Currently, text-to-text is available. But image embedders are not, as far as I could read in the documentation and blog.
Blog examples require an external Python code to produce the image vectors.

**Comment by jobergum** [comment]
Makes sense. CLIP has two parts, image encoding and text encoding, and are handled by two different neural networks. 

We could fit the text transformer model into the existing embed framework as already done in multiple vespa sample applications, but image encoding would not fit into the existing embed functionality which takes a string or array of string as inputs.

**Comment by jobergum** [comment]
So if you are fine with just having the text-to-image space model in Vespa, we can create that type of example using HF-embedder functionality.

**Comment by eostis** [comment]
With the same process ?

- Export the HF CLIP .onnx
- Set the containers's HF component in services.xml
- Define the input fields and input images embedding in .sd
- Add a closeness rank profile
- Define the YQL query with nearestNeighbour and the ranking

**Comment by jobergum** [comment]
To handle image data, we would have to create a new type of embedder functionality.

**Comment by eostis** [comment]
Exactly! It will also prepare Vespa for further types: audio, video ...

**Comment by eostis** [comment]
I was a bit ahead of time apparently. [7-modality](https://weaviate.io/developers/weaviate/modules/retriever-vectorizer-modules/multi2vec-bind) is here.

**Comment by jobergum** [comment]
ImageBind is interesting, but I do recommend looking at the licensing :)

**Comment by jobergum** [comment]
Hey @AriMKatz, 

We currently do not expose any provided [embedders](https://docs.vespa.ai/en/embedding.html#provided-embedders) that is for multimodal. The provided embedder models are text only. 

This doesn't mean that you cannot use multimodal representations with Vespa, for example here is a recent example of a multimodal model [PDF Retrieval with Vision Language Models (ColPali)](https://blog.vespa.ai/retrieval-with-vision-language-models-colpali/).

**Comment by alpha-javed** [comment]
Hey @jobergum,
does vespa support multimodality currently?

**Comment by jobergum** [comment]
See my comment above and https://github.com/vespa-engine/vespa/issues/32389, the native built in embedders are currently only text.
